File: Server/ClientSession.cs
using System;
using System.Net;
using ServerCore;
using Google.Protobuf.Protocol;

namespace Server
{
    class ClientSession : PacketSession
    {
        public override void OnConnected(EndPoint endPoint)
        {
            Console.WriteLine($"OnConnected : {endPoint}");

        }

        public override void OnReceivePacket(ArraySegment<byte> buffer)
        {
            ServerPacketManager.Instance.OnRecvPacket(this, buffer);
        }

        public override void OnSend(int numberOfBytes)
        {
            
        }
        public override void OnDisconnected(EndPoint endPoint)
        {
            Console.WriteLine($"OnDisconnected : {endPoint}");
        }

    }
}

File: Server/ClientSession.cpp
#include "pch.h"
#include "ClientSession.h"
#include "SessionManager.h"	  
#include <SendBufferManager.h>	


void ClientSession::OnConnected()
{
	SessionManager::Get().Add(static_pointer_cast<ClientSession>(shared_from_this()));
}
int ClientSession::OnRecvPacket(BYTE* buffer, int len)
{
	shared_ptr<SendBuffer> sendBuffer = SendBufferManager::Get().Open(4096);
	memcpy(sendBuffer->GetBuffer(), buffer, len);

	if (sendBuffer->Close(len))
	{
		Send(sendBuffer);
	}

	printf("%s\n", &buffer[4]);
	return len;
}
void ClientSession::OnSend(int len)
{
}
void ClientSession::OnDisconnected()
{
	SessionManager::Get().Remove(static_pointer_cast<ClientSession>(shared_from_this()));
}
